src/backends/multicore/intrin_avx512.hpp

```diff
@@ -0,0 +1,80 @@
+#pragma once
+
+// vector types for avx512
+
+// double precision avx512 register
+using vecd_avx512  = __m512d;
+// 8 way mask for avx512 register (for use with double precision)
+using mask8_avx512 = __mmask8;
+
+inline vecd_avx512 set(double x) {
+    return _mm512_set1_pd(x);
+}
+
+namespace detail {
+    // Useful constants in vector registers
+    const vecd_avx512 vecd_avx512_zero = set(0.0);
+    const vecd_avx512 vecd_avx512_one  = set(1.0);
+    const vecd_avx512 vecd_avx512_two  = set(2.0);
+    const vecd_avx512 vecd_avx512_nan  = set(std::numeric_limits<double>::quiet_NaN());
+    const vecd_avx512 vecd_avx512_inf  = set(std::numeric_limits<double>::infinity());
+    const vecd_avx512 vecd_avx512_ninf = set(-std::numeric_limits<double>::infinity());
+}
+
+//
+// Operations on vector registers.
+//
+// shorter, less verbose wrappers around intrinsics
```

Comment:
I like the wrappers. I think we should template them on the backend (AVX2, AVX512 etc.) and expose them to the upper levels.

Reply:
The wrappers over simple operations like `add` don't need to be templated I think.
You can just use function overloading to do the trick (as opposed to having to provide specializations for different vector registers).